src/message.c

```diff
@@ -1223,118 +1223,124 @@ wait_return(int redraw)
+			    got_int = FALSE;
+			}
+			else if (c != K_IGNORE)
+			{
+			    c = K_IGNORE;
+			    hit_return_msg();
+			}
 		    }
-		    else if (c != K_IGNORE)
-		    {
+		    else if (msg_scrolled > Rows - 2
+			    && (c == 'j' || c == 'd' || c == 'f'
+					    || c == K_DOWN || c == K_PAGEDOWN))
 			c = K_IGNORE;
-			hit_return_msg();
-		    }
 		}
-		else if (msg_scrolled > Rows - 2
-			 && (c == 'j' || c == 'd' || c == 'f'
-					   || c == K_DOWN || c == K_PAGEDOWN))
-		    c = K_IGNORE;
-	    }
-	} while ((had_got_int && c == Ctrl_C)
-				|| c == K_IGNORE
+	    } while ((had_got_int && c == Ctrl_C)
+				    || c == K_IGNORE
 #ifdef FEAT_GUI
-				|| c == K_VER_SCROLLBAR || c == K_HOR_SCROLLBAR
+				    || c == K_VER_SCROLLBAR || c == K_HOR_SCROLLBAR
 #endif
-				|| c == K_LEFTDRAG   || c == K_LEFTRELEASE
-				|| c == K_MIDDLEDRAG || c == K_MIDDLERELEASE
-				|| c == K_RIGHTDRAG  || c == K_RIGHTRELEASE
-				|| c == K_MOUSELEFT  || c == K_MOUSERIGHT
-				|| c == K_MOUSEDOWN  || c == K_MOUSEUP
-				|| c == K_MOUSEMOVE
-				|| (!mouse_has(MOUSE_RETURN)
-				    && mouse_row < msg_row
-				    && (c == K_LEFTMOUSE
-					|| c == K_MIDDLEMOUSE
-					|| c == K_RIGHTMOUSE
-					|| c == K_X1MOUSE
-					|| c == K_X2MOUSE))
-				);
-	ui_breakcheck();
-
-	// Avoid that the mouse-up event causes Visual mode to start.
-	if (c == K_LEFTMOUSE || c == K_MIDDLEMOUSE || c == K_RIGHTMOUSE
-					  || c == K_X1MOUSE || c == K_X2MOUSE)
-	    (void)jump_to_mouse(MOUSE_SETPOS, NULL, 0);
-	else if (vim_strchr((char_u *)"\r\n ", c) == NULL && c != Ctrl_C)
-	{
-	    // Put the character back in the typeahead buffer.  Don't use the
-	    // stuff buffer, because lmaps wouldn't work.
-	    ins_char_typebuf(vgetc_char, vgetc_mod_mask);
-	    do_redraw = TRUE;	    // need a redraw even though there is
-				    // typeahead
+				    || c == K_LEFTDRAG   || c == K_LEFTRELEASE
+				    || c == K_MIDDLEDRAG || c == K_MIDDLERELEASE
+				    || c == K_RIGHTDRAG  || c == K_RIGHTRELEASE
+				    || c == K_MOUSELEFT  || c == K_MOUSERIGHT
+				    || c == K_MOUSEDOWN  || c == K_MOUSEUP
+				    || c == K_MOUSEMOVE
+				    || (!mouse_has(MOUSE_RETURN)
+					&& mouse_row < msg_row
+					&& (c == K_LEFTMOUSE
+					    || c == K_MIDDLEMOUSE
+					    || c == K_RIGHTMOUSE
+					    || c == K_X1MOUSE
+					    || c == K_X2MOUSE))
+				    );
+	    ui_breakcheck();
+
+	    // Avoid that the mouse-up event causes Visual mode to start.
+	    if (c == K_LEFTMOUSE || c == K_MIDDLEMOUSE || c == K_RIGHTMOUSE
+					    || c == K_X1MOUSE || c == K_X2MOUSE)
+		(void)jump_to_mouse(MOUSE_SETPOS, NULL, 0);
+	    else if (vim_strchr((char_u *)"\r\n ", c) == NULL && c != Ctrl_C)
+	    {
+		// Put the character back in the typeahead buffer.  Don't use the
+		// stuff buffer, because lmaps wouldn't work.
+		ins_char_typebuf(vgetc_char, vgetc_mod_mask);
+		do_redraw = TRUE;	    // need a redraw even though there is
+					// typeahead
+	    }
+	} else {
+	    c = CAR;
+	    do_sleep(200L, TRUE);
```

Comment:
Why does it sleep 200ms? Please write comments of the reason at this and docs.

Reply:
I have added the documentation.